### PR TITLE
Pz/types 004 - Add missing Customers API endpoints

### PR DESCRIPTION
**Pull request by marcin-slezak**

### CODE REVIEW

**Comment by weronika-kurczyna** [comment]
fixes:
- Updated naming of request and responses
- `CustomersUpdateCustomersInBulkRequestBody` was missing `birthdate?: string` (it is possible to update birthdate of customers)
- there was a typo i `data_json` : daily_event instead od daily_events
remark:
endpoint which deletes customer's data permanently has a typo, I tested it in postman, correctly it should return `object: 'permanent deletion'`, but in fact it returns `object: 'pernament_deletion'`

**Comment by weronika-kurczyna** [comment]
i think it's still minor, the endpoints and interfaces are completely new, and i only changed naming of the interfaces added by Piotr

File: packages/sdk/src/types/Customers.ts [hunk 1]
```diff
@@ -136,3 +137,55 @@ export type CustomersUpdateParams = CustomerRequest & IdOrSourceId
 export type CustomersUpdateResponse = CustomerObject | CustomerUnconfirmed
 
 export type CustomersUpdateConsentsBody = Record<string, boolean>
+
+// New types following guideline
+
+// 0-level types
```
Comment (marcin-slezak) [inline packages/sdk/src/types/Customers.ts:143]: `CustomerBase` and `CustomerAddress` are Domain Types
0-level types are `CustomersUpdateInBulkRequestBody` , `CustomersUpdateMetadataInBulkRequestBody` and `CustomersDeletePermanentlyResponseBody`

File: packages/sdk/src/types/Customers.ts [hunk 2]
```diff
@@ -136,3 +137,55 @@ export type CustomersUpdateParams = CustomerRequest & IdOrSourceId
...
+		country?: string | null
+		postal_code?: string | null
+	} | null
+}
+
+export type CustomersUpdateInBulkRequestBody = CustomerBase &
+	CustomerAddress &
+	{
+		source_id: string
+	}[]
```
Comment (marcin-slezak) [inline packages/sdk/src/types/Customers.ts:169]: As far as I see, the `CustomersUpdateInBulkRequestBody` type is not defined correctly. It should describe an array of objects like:

```ts
const someRequest:CustomersUpdateInBulkRequestBody = [
	{name: 'asdasd', source_id: '1234asd'}
]

```

but it does not. That's the reason for the TS error in tests:

image


The type should look like 

```ts
export type CustomersUpdateInBulkRequestBody = (
		CustomerBase &
		CustomerAddress &
		{
			source_id: string
		}
	)[]
```

Comment (weronika-kurczyna) [reply packages/sdk/src/types/Customers.ts:169]: Interestingly, my idea did not display this error and tests are passing 🫣

**Comment by weronika-kurczyna** [comment]
image
it is allowed to send `source_id` as null :o

File: packages/sdk/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+// Domain types 👇
+
+// Basic customer properties, common for most of the 0-level endpoints
+type CustomerBase = {
+	name?: string
+	email?: string
+	phone?: string
+	description?: string
+	birthdate?: string
+	metadata?: Record<string, any>
```
Comment (marcin-slezak) [inline packages/sdk/CONTRIBUTING.md:88]: Change "any" to "unknown" -> you still can assign here anything, but it does not turn off all TS safeguards when you read it

File: packages/sdk/src/types/Customers.ts [hunk 4]
```diff
@@ -136,3 +137,56 @@ export type CustomersUpdateParams = CustomerRequest & IdOrSourceId
...
+
+// Domain types
+
+type CustomerBase = {
+	name?: string | null
+	description?: string | null
+	email?: string | null
+	phone?: string | null
+	birthdate?: string | null
+	metadata?: Record<string, any> | null
```
Comment (marcin-slezak) [inline packages/sdk/src/types/Customers.ts:151]: Change "any" to "unknown" -> you still can assign here anything, but it does not turn off all TS safeguards when you read it

**Pull request merged**